solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -268,89 +307,195 @@ contract InterchainAccountRouterTest is Test {
+            calls[0].data,
+            ""
         );
+
+        // assert
+        uint256 balanceAfter = address(this).balance;
         assertRemoteCallReceived(data);
+        assertIgpPayment(balanceBefore, balanceAfter, igp.getDefaultGasUsage());
     }
 
-    function testCallRemoteWithDefault(bytes32 data) public {
+    function testFuzz_callRemoteWithDefault(bytes32 data) public {
+        // arrange
         originRouter.enrollRemoteRouterAndIsm(
             destination,
             routerOverride,
             ismOverride
         );
-        originRouter.callRemote(destination, getCalls(data));
+        uint256 balanceBefore = address(this).balance;
+
+        // act
+        originRouter.callRemote{value: gasPaymentQuote}(
+            destination,
+            getCalls(data)
+        );
+
+        // assert
+        uint256 balanceAfter = address(this).balance;
         assertRemoteCallReceived(data);
+        assertIgpPayment(balanceBefore, balanceAfter, igp.getDefaultGasUsage());
     }
 
-    function testOverrideAndCallRemote(bytes32 data) public {
+    function testFuzz_overrideAndCallRemote(bytes32 data) public {
+        // arrange
         originRouter.enrollRemoteRouterAndIsm(
             destination,
             routerOverride,
             ismOverride
         );
-        originRouter.callRemote(destination, getCalls(data));
+        uint256 balanceBefore = address(this).balance;
+
+        // act
+        originRouter.callRemote{value: gasPaymentQuote}(
+            destination,
+            getCalls(data)
+        );
+
+        // assert
+        uint256 balanceAfter = address(this).balance;
         assertRemoteCallReceived(data);
+        assertIgpPayment(balanceBefore, balanceAfter, igp.getDefaultGasUsage());
     }
 
-    function testCallRemoteWithoutDefaults(bytes32 data) public {
+    function testFuzz_callRemoteWithoutDefaults_revert_noRouter(
+        bytes32 data
+    ) public {
+        // assert error
         CallLib.Call[] memory calls = getCalls(data);
         vm.expectRevert(bytes("no router specified for destination"));
         originRouter.callRemote(destination, calls);
     }
 
-    function testCallRemoteWithOverrides(bytes32 data) public {
-        originRouter.callRemoteWithOverrides(
+    function testFuzz_customMetadata_forIgp(
+        uint64 gasLimit,
+        uint64 overpayment,
+        bytes32 data
+    ) public {
+        // arrange
+        bytes memory metadata = StandardHookMetadata.formatMetadata(
+            0,
+            gasLimit,
+            address(this),
+            ""
+        );
+        originRouter.enrollRemoteRouterAndIsm(
+            destination,
+            routerOverride,
+            ismOverride
+        );
+        uint256 balanceBefore = address(this).balance;
+
+        // act
+        originRouter.callRemote{value: gasLimit * igp.gasPrice() + overpayment}(
+            destination,
+            getCalls(data),
+            metadata
+        );
```

Comment:
should we test underpayment with hook metadata too? as it stands these arent really unit tests